org.eclipse.scanning.api/src/org/eclipse/scanning/api/filter/Filter.java

```diff
@@ -138,7 +141,17 @@ public boolean equals(Object obj) {
 		Collection<String> excludes = match(getExcludes(), items);
 		ret.removeAll(excludes);
 		Collection<String> includes = match(getIncludes(), items);
-		ret.addAll(includes);
+		
+		Collection<String> done = new HashSet<>();
+		for (final String item : includes) {
+			
+			if (done.contains(item)) continue;

```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') At most one statement is allowed per line, but 2 statements were found on this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00122)